lib/spanner/config/upgrader.ex

```diff
@@ -0,0 +1,130 @@
+defmodule Spanner.Config.Upgrader do
+  alias Spanner.Config.SyntaxValidator
+
+  @moduledoc """
+  Attempts to upgrade bundle config to current version.
+  """
+
+  @current_version 3
+  @upgradable_versions [2]
```

Comment:
I think this can be a single integer since we're only planning to support N and N - 1 for now.